### PR TITLE
Mat 7155 shift test caes dates

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-7155](https://jira.cms.gov/browse/MAT-7155)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+      List<DataElement> elements = testCaseJson.getDataElements();
+      if (CollectionUtils.isNotEmpty(elements)) {
+        for (DataElement element : elements) {
+          shiftDates(element, shifted);
+        }
+      }
+      String newJson = mapper.writeValueAsString(testCaseJson);
+      testCase.setJson(newJson);
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: can we log this instead?

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+      throw new CqmConversionException(
+          "JsonProcessingException for test case id : " + testCase.getId());
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: and this

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    }
+
+    shiftDatesForTestCase(testCase, shifted, measureId, username, accessToken);
+    testCaseService.updateTestCase(testCase, measureId, username, accessToken);
+
+    return testCase;
+  }
+
+  protected TestCase shiftDatesForTestCase(
+      TestCase testCase, int shifted, String measureId, String username, String accessToken) {
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: I think a couple of these parameters are unused

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+      }
+      String newJson = mapper.writeValueAsString(testCaseJson);
+      testCase.setJson(newJson);
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
+      throw new CqmConversionException(
+          "JsonProcessingException for test case id : " + testCase.getId());
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new CqmConversionException("Exception for test case id : " + testCase.getId());
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: A more elaborative exception message would help to understand when "`Exception for test case id`" occurred. maybe something like "`An issue occurred while shifting the test case dates for the test case...`". same at line 119.

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: @adongare Thanks, this is done.

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: @adongare done.

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java (hunk 3 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: @adongare Yeah it was due to a code refactor, I removed the unused parameters, thanks!

File: src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java (hunk 4 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java]: @adongare done.

**Review by adongare — APPROVED** [review]

**Pull request merged**